reframe/core/pipeline.py

```diff
@@ -681,6 +691,11 @@ def __new__(cls, *args, **kwargs):
     def __init__(self):
         pass
 
+    @classmethod
+    def __init_subclass__(cls, *, extended_test=False, **kwargs):
```

Comment:
I think I like more the `special` keyword, since it's a test treated specially:

```suggestion
    def __init_subclass__(cls, *, special=False, **kwargs):
```